apps/native-component-list/src/screens/ApplicationScreen.tsx

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import { ScrollView, View, Text, Button, Platform } from 'react-native';
+import * as Application from 'expo-application';
+import HeadingText from '../components/HeadingText';
+import MonoText from '../components/MonoText';
+
+interface ApplicationConstantsName {
+  name?: string;
+  value?: any;
+}
+
+interface State {
+  installReferrer?: string;
+  idForVendor?: string;
+  firstInstallDate?: string | undefined;
+  lastUpdateDate?: string | undefined;
+}
+
+class ApplicationConstants extends React.Component<ApplicationConstantsName> {
+  render() {
+    let { name, value } = this.props;
+    if (typeof value === 'boolean') {
+      return (
+        <View style={{ marginBottom: 10 }}>
+          <HeadingText>{name}</HeadingText>
+          <MonoText> {String(value)}</MonoText>
+        </View>
+      );
+    } else {
+      return (
+        <View style={{ marginBottom: 10 }}>
+          <HeadingText>{name}</HeadingText>
+          <MonoText> {value}</MonoText>
+        </View>
+      );
+    }
+  }
+}
+
+export default class ApplicationScreen extends React.Component<{}, State> {
+  static navigationOptions = {
+    title: 'Application',
+  };
+
+  state: State = {
```

Comment:
React should infer that it's of type `State` if you pass proper generic type to `React.Component<Props, State>` 🙂